src/linker/Linker/MessageContainer.cs

```diff
@@ -132,6 +132,16 @@ private static MessageContainer CreateWarningMessageContainer (LinkContext conte
 			if (version > context.WarnVersion)
 				return Empty;
 
+			if (origin.MemberDefinition != null) {
```

Comment:
How many warnings have origin.MemberDefinition == null?

Reply:
It's a little hard to tell because we pass around the source location through multiple helpers, but I believe the trim analysis warnings will have a MemberDefinition. We don't have a source location for assembly-level attributes, but I can't think of a case where we can produce dataflow warnings from these.

I made the check more specific (checking the subcategory) and added an assert for this.